source/MaterialXGenShader/GenContext.h

```diff
@@ -107,15 +107,19 @@ class GenContext
         return _sourceCodeSearchPath.find(filename);
     }
 
-    /// Add the given name to the list of unique identifiers in use.
-    void addIdentifier(const string& name);
-
-    /// Make sure the given name is a unique identifiers, 
-    /// updating the name if needed.
-    void makeIdentifier(string& name);
```

Comment:
Moved identifier generation to the Syntax class where it belongs. In the context we just keep a record of reserved words.